### ISSUE TITLE
Content-Type: application/json; charset=utf-8 results in "Could not find deserializer"

### ISSUE THREAD

**Issue by MarZl** [googleforgames/agones#1748]
**What happened**:
Hi, I'm still new to k8s and agones, so it might be my fault.. I'm using the Java Generic Kubernetes API and created/generated the classes for the agones CRDs. Reading the resources was fine, but creating new Resources failed. This is from the agones-allocator:

``` 
{
  "error": "Could not find deserializer",
  "headers": {
    "Accept": [
      "application/json"
    ],
    "Accept-Encoding": [
      "gzip"
    ],
    "Content-Length": [
      "186"
    ],
    "Content-Type": [
      "application/json; charset=utf-8"
    ],
    "User-Agent": [
      "OpenAPI-Generator/1.0-SNAPSHOT/java"
    ],
    "X-Forwarded-For": [
      "192.168.178.136"
    ],
    "X-Forwarded-Host": [
      "10.99.197.45:443"
    ],
    "X-Forwarded-Proto": [
      "https"
    ],
    "X-Forwarded-Uri": [
      "/apis/allocation.agones.dev/v1/namespaces/mynamespace/gameserverallocations"
    ],
    "X-Remote-Group": [
      "system:masters",
      "system:authenticated"
    ],
    "X-Remote-User": [
      "minikube-user"
    ]
  },
  "host": "10.99.197.45:443",
  "message": "",
  "method": "POST",
  "requestURI": "/apis/allocation.agones.dev/v1/namespaces/mynamespace/gameserverallocations",
  "severity": "error",
  "source": "*apiserver.APIServer",
  "time": "2020-08-12T18:56:23.905144673Z",
  "url": {
    "Scheme": "",
    "Opaque": "",
    "User": null,
    "Host": "",
    "Path": "/apis/allocation.agones.dev/v1/namespaces/mynamespace/gameserverallocations",
    "RawPath": "",
    "ForceQuery": false,
    "RawQuery": "",
    "Fragment": ""
  }
}
{
  "message": "Could not find deserializer",
  "severity": "error",
  "stack": [
    "agones.dev/agones/pkg/gameserverallocations.(*Controller).allocationDeserialization\n\t/go/src/agones.dev/agones/pkg/gameserverallocations/controller.go:180",
    "agones.dev/agones/pkg/gameserverallocations.(*Controller).processAllocationRequest\n\t/go/src/agones.dev/agones/pkg/gameserverallocations/controller.go:131",
    "agones.dev/agones/pkg/gameserverallocations.(*Controller).registerAPIResource.func1\n\t/go/src/agones.dev/agones/pkg/gameserverallocations/controller.go:93",
    "agones.dev/agones/pkg/util/apiserver.(*APIServer).resourceHandler.func1\n\t/go/src/agones.dev/agones/pkg/util/apiserver/apiserver.go:152",
    "agones.dev/agones/pkg/util/https.ErrorHTTPHandler.func1\n\t/go/src/agones.dev/agones/pkg/util/https/https.go:52",
    "net/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2012",
    "net/http.(*ServeMux).ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2387",
    "net/http.serverHandler.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2807",
    "net/http.initALPNRequest.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:3381",
    "net/http.(*http2serverConn).runHandler\n\t/usr/local/go/src/net/http/h2_bundle.go:5720",
    "runtime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1373"
  ],
  "time": "2020-08-12T18:56:23.905438557Z"
}
```

**What you expected to happen**:
It should work with 'Content-Type: application/json; charset=utf-8' 

**How to reproduce it (as minimally and precisely as possible)**:
Create a fleet for namespace mynamespace as in the guide but try to allocate via the REST api instead using kubectl.
```
APISERVER=https://kubernetes.default.svc
SERVICEACCOUNT=/var/run/secrets/kubernetes.io/serviceaccount
TOKEN=$(cat ${SERVICEACCOUNT}/token)
CACERT=${SERVICEACCOUNT}/ca.crt

curl --cacert ${CACERT} --header "Authorization: Bearer ${TOKEN}" -X POST -H 'Content-Type: application/json; charset=utf-8' --data '
{
   "apiVersion": "allocation.agones.dev/v1",
   "kind": "GameServerAllocation",
   "spec": {
      "required": {
         "matchLabels": {
            "agones.dev/fleet": "simple-udp"
         }
      }
   }
}
' ${APISERVER}/apis/allocation.agones.dev/v1/namespaces/mynamespace/gameserverallocations
```

**Anything else we need to know?**:
I tried to find where the "; charset=utf-8" is getting added. I debugged the the java client, but in there it's always just "application/json". It might be added later.

**Environment**:
- Agones version: 
agones-1.7.0

- Kubernetes version (use `kubectl version`):
Minikube
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.6", GitCommit:"dff82dc0de47299ab66c83c626e08b245ab19037", GitTreeState:"clean", BuildDate:"2020-07-15T16:58:53Z", GoVersion:"go1.13.9", Compiler:"gc", Plat form:"windows/amd64"}                                                                                                                                                                                                             Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.2", GitCommit:"59603c6e503c87169aea6106f57b9f242f64df89", GitTreeState:"clean", BuildDate:"2020-01-18T23:22:30Z", GoVersion:"go1.13.5", Compiler:"gc", Plat form:"linux/amd64"}

- Cloud provider or hardware configuration:
VirtualBox

- Install method (yaml/helm):
helm v3.2.4

- Troubleshooting guide log(s):
- Others:
Thanks!

**Comment by markmandel** [comment]
Ooh! That's a legit fun one! We do some custom stuff on the allocation path that may not handle that header properly:
https://github.com/googleforgames/agones/blob/master/pkg/gameserverallocations/controller.go#L178

(Looks like it doesn't).

As a workaround, you can use the Allocation Service, which has a gRPC endpoint - which runs exactly the same code:
https://agones.dev/site/docs/advanced/allocator-service/

**Comment by markmandel** [comment]
This can be solved with: `mime.ParseMediaType(...)`

We can see it in action here: https://play.golang.org/p/aGc-e8OkLAo

**Comment by aLekSer** [comment]
Just found a bit easier way to reproduce it. (Initial option was to run `kubectl run curl-1 --image=radial/busyboxplus:curl -i --tty --rm`)
```
APISERVER=http://localhost:8001
SERVICEACCOUNT=/var/run/secrets/kubernetes.io/serviceaccount

curl -X POST -H 'Content-Type: application/json; charset=utf-8' --data '
{
   "apiVersion": "allocation.agones.dev/v1",
   "kind": "GameServerAllocation",
   "spec": {
      "required": {
         "matchLabels": {
            "agones.dev/fleet": "simple-udp"
         }
      }
   }
}
' ${APISERVER}/apis/allocation.agones.dev/v1/namespaces/allocation/gameserverallocations

Could not find deserializer
```

**Comment by MarZl** [comment]
Awesome! Thanks for the fast response and fix.

**Comment by markmandel** [comment]
This will be included in the 1.8.0 release that will come out next Tuesday :+1:
